agent/engine/task_manager.go

```diff
@@ -407,6 +407,11 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 	if event.Status <= containerKnownStatus {
 		seelog.Infof("Managed task [%s]: redundant container state change. %s to %s, but already %s",
 			mtask.Arn, container.Name, event.Status.String(), containerKnownStatus.String())
+
+		// Only update container metadata when status stays RUNNING
+		if event.Status == containerKnownStatus && event.Status == apicontainerstatus.ContainerRunning {
```

Comment:
I wonder if we are updating the container metadata without the container state change, should we do this checking "if event.Status == containerKnownStatus" independently outside this "if" block and update the log with this change? Also if we only update container metadata when status stays RUNNING, maybe checking "event.Status == apicontainerstatus.ContainerRunning" before "event.Status == containerKnownStatus" makes more sense?

Reply:
I don't really wanna have repetitive code for logging, that's why the if is inside the block. And what's the reason behind checking "event.Status == apicontainerstatus.ContainerRunning" first? I think it does not really make a big difference.